### DISCUSSION TITLE
API Proxying for Dev doesn't work in BEX mode

### DISCUSSION THREAD

**Discussion by oussama-he** [quasarframework/quasar#9434] [CLI - BEX mode]
I'm building a project using Quasar, and I want to create a browser extension that interacts with a backend. When I trying to use API endpoint of the backend I get this error: 
```
Failed to load resource: net::ERR_FILE_NOT_FOUND
```
![Screenshot at 2021-05-26 21-24-32](https://user-images.githubusercontent.com/18425315/119726877-8bba0680-be69-11eb-9044-324deb334761.png)

Although I'm using API Proxying, and below how I config it in `quasar.conf.js`:
```js
devServer: {
      https: false,
      port: 8080,
      open: false, // opens browser window automatically
      proxy: [{
        // proxy all requests starting with /api
        context: ['/api'],
          target: 'http://localhost:8888',
      }]
    },
```
For your information, I get used the endpoint outside of the browser extension without errors [it worked in `pages/Index.vue`] 

Please, how can I fix this?

Thank you in advance.

**Comment by ontwikkelfabriek** [comment]
@oussama-he 
You do know that the `devServer` and all it's settings only apply to `dev` mode? So it won't proxy anything for you in production.

Inspecting your image I see `api/v1/core/tweets/stats/:1`
However in the proxy you use `/api` with a slash.

Looking at the Quasar docs  about setting up a proxy I would try:
```javascript
api: {
      target: 'http://localhost:8888',
      changeOrigin: true,
      pathRewrite: {
        '^/api': '' // or `api` instead of `/api`
      }
}
```

Here's the Quasar doc about ajax request's in bex mode.
 https://quasar.dev/quasar-cli/ajax-requests   

btw did you try it without using a proxy? ( proxy's are generally used to bypass cors issues in dev mode)
